python/src/scipp/compat/mantid.py

```diff
@@ -161,9 +161,127 @@ def convert_EventWorkspace_to_dataset(ws, load_pulse_times, EventType):
+        ws : Mantid TableWorkspace
+            Mantid TableWorkspace to be converted into scipp dataset
+
+    Keyword arguments
+    -----------------
+        error_connection : Dict
+            Dict with data column names as keys to names of their error column
+    """
+
+    # Extract information from workspace
+    n_columns = ws.columnCount()
+    columnNames = ws.getColumnNames()  # list of names matching each column
+    columnTypes = ws.columnTypes()  # list of types matching each column
+
+    if error_connection is None:
+        # Generate error connection
+        error_connection = {}
+        # Dictionary with data column keys for names of their error columns
+        # Uses the plottype that is built into TableWorkspace
+        plottype_list = []
+        for i in range(n_columns):
+            plottype_list.append(ws.getPlotType(i))
+
+        # Relevant plottype indices
+        X_index = 1
+        Xerr_index = 4
+        Y_index = 2
+        Yerr_index = 5
+
+        if plottype_list.count(Xerr_index) > 1:  # More than one Xerr
+            raise RuntimeError(
+                "Multiple Xerr columns in TableWorkspace, error_connection "
+                "need to be specified manually for the scipp "
+                "convert_TableWorkspace_to_dataset function.")
+        elif plottype_list.count(Xerr_index) == 1:
+            error_index = plottype_list.index(Xerr_index)
+            errorName = columnNames[error_index]
+            if plottype_list.count(X_index) == 1:  # Unique X for Xerr
+                data_index = plottype_list.index(X_index)
+                data_name = columnNames[data_index]
+                error_connection[data_name] = errorName
+            else:
+                raise RuntimeError(
+                    "No unique X data to be matched with Xerr column, "
+                    "error_connection need to be specified manually for the "
+                    "scipp convert_TableWorkspace_to_dataset function.")
+
+        if plottype_list.count(Yerr_index) > 1:  # More than one Yerr
+            raise RuntimeError(
+                "Multiple Yerr columns in TableWorkspace, error_connection "
+                "need to be specified manually for the scipp "
+                "convert_TableWorkspace_to_dataset function.")
+        elif plottype_list.count(Yerr_index) == 1:
+            error_index = plottype_list.index(Yerr_index)
+            errorName = columnNames[error_index]
+            if plottype_list.count(Y_index) == 1:  # Unique Y for Yerr
+                data_index = plottype_list.index(Y_index)
+                data_name = columnNames[data_index]
+                error_connection[data_name] = errorName
+            else:
+                raise RuntimeError(
+                    "No unique Y data to be matched with Yerr column, "
+                    "error_connection need to be specified manually for the "
+                    "scipp convert_TableWorkspace_to_dataset function.")
```

Comment:
These 15 lines of code appear twice, near-identical. Can it be made a function?

Reply:
This code was deleted as plottype is now ignored.